test/report.test.ts

```diff
@@ -0,0 +1,55 @@
+// (C) 2007-2018 GoodData Corporation
+import 'isomorphic-fetch';
+import * as fetchMock from 'fetch-mock';
+import { ReportModule } from '../src/report';
+import { XhrModule, ApiResponseError } from '../src/xhr';
+import { ExportConfigFormat } from '../src/interfaces';
+import { IExportResponse } from '../src/apiResponsesInterfaces';
+
+const mockedReportModule = () => new ReportModule(new XhrModule(fetch, {}));
+
+describe('report', () => {
+    const projectId = '1';
+    const projectUri = `/gdc/internal/projects/${projectId}/exportResult`;
+
+    describe('export', () => {
+        afterEach(() => {
+            fetchMock.restore();
+        });
+
+        describe('exportResult', () => {
+            it('should return created file', () => {
+                const createdReport = '/gdc/exporter/result/12345';
+                fetchMock.mock(
+                    projectUri,
+                    {
+                        status: 200,
+                        body: { uri: createdReport }
+                    }
+                );
+
+                const executionResult = '/executionResult/1234';
+                const exportConfig = {
+                    title: 'title',
+                    format: 'xlsx' as ExportConfigFormat,
```

Comment:
this `as "Re-typing"` is not necessary :) just you have to specify type of whole object

```
const exportConfig: IExportConfig = {
	title: 'title',
    format: 'xlsx',
	mergeHeaders: false,
    includeFilterContext: true
};
```